src/View/SSViewer_DataPresenter.php

```diff
@@ -40,95 +40,126 @@ class SSViewer_DataPresenter extends SSViewer_Scope
 
     /**
      * Underlay variables. Concede precedence to overlay variables or anything from the current scope
-     * @var array|null
+     *
+     * @var array
      */
     protected $underlay;
 
-    public function __construct($item, $overlay = null, $underlay = null, $inheritedScope = null)
-    {
+    /**
+     * @var object $item
+     * @var array $overlay
+     * @var array $underlay
+     * @var SSViewer_Scope $inheritedScope
+     */
+    public function __construct(
+        $item,
+        array $overlay = null,
+        array $underlay = null,
+        SSViewer_Scope $inheritedScope = null
```

Comment:
Typehinting added here, not part of our “public” API but happy to undo this if others disagree